Support Django<=1.11 (was: Migrate to parser.add_arguments())

thor opened the pull request:
Uses parser.add_arguments() as the previous approach has been deprecated
since 1.8.

Fixes #59.

A possibly downside is that [with the change](https://docs.djangoproject.com/en/1.10/releases/1.8/#extending-management-command-arguments-through-command-option-list), I have not implemented a solution to allow Django<1.8 to work. 

*Secondary thoughts*: perhaps it'd be better to stop supporting 1.7, seeing as it is not a supported version anymore, and furthermore add test environments for 1.10 and 1.11 while at it?

*Ternary thoughts*: Should this PR be a PR to remove support for 1.7, whilst adding it for 1.10 and 1.11?

willkg commented:
I did some poking around regarding Django support. This is the list of projects that I can find that use Puente:

* mozilla/mozillians
* mozilla/kuma
* mdn/browsercompat
* mozilla/addons-server
* mozilla/bedrock
* mozilla/kitsune
* mozilla/fjord

The last two are decommissioned--so we can ignore them. The first four are all using Django 1.8.

Given that, I think the following is fine:

1. we drop support for Django 1.7 -- I'll do that in a separate PR
2. we add support for Django 1.9 and 1.10 and 1.11-beta if you want to do the work
3. this PR turns into a "support Django >= 1.9" PR and covers whatever issues exist

Does that sound ok?

thor commented:
Sounds good to me! :)

You mention dropping support for Django 1.7 in a separate PR -- would you like me to leave the test environment for Django 1.7 in [tox.ini](https://github.com/mozilla/puente/blob/7e37132eded96bf018b9e842ffbea5e4061e4853/tox.ini), or can I remove it whilst adding 1.9, 1.10 and 1.11?

I'm okay with adding the tests for 1.10 and 1.11 to see to it that it all works. (I'm currently running a modified version of puente with 1.10 in production, which works, although I have a documentation suggestion in regards to extension load order.)

willkg commented:
I don't think anyone really uses this project, so I don't think I'm going to worry about anyone else at the moment.

I'm game for sharing the work here. What's the easiest path of changes for you?

willkg commented:
@thor ^^^ Any preference?

thor commented:
Sorry, busy days @willkg!
I think it'd be fairly easy to add the test configuration, and make any
necessary changes, so I don't mind that.

Either path's approachable either way. :+1:

1. mar. 2017 02:10 skrev "Will Kahn-Greene" <notifications@github.com>:

@thor <https://github.com/thor> ^^^ Any preference?

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/mozilla/puente/pull/60#issuecomment-283213197>, or mute
the thread
<https://github.com/notifications/unsubscribe-auth/AAOT-WFzQuDHbQRCT0nm0HQRM2tESxMKks5rhMWIgaJpZM4MMesd>
.

willkg commented:
I landed PR #61 which drops support for jingo and django 1.7 and adds support for python 3.5 and 3.6.

Can you rebase this branch on top of that? That should covers the Django 1.7 issues.

thor commented:
@willkg rebased! I changed the tox setup to only use Python 3.6 for Django 1.11, as it is the only version of Django that supports 3.6. Furthermore, I changed the ´tox.ini´ to use conditonal settings for easier overview. As a small addendum, I edited ´travis.yml´ so that it would have the necessary environments, but also only run the relevant tests for the relevant environment.

If it's a bit too out of scope for the PR, I can remove it, but it seems to tie it up fairly well, I think. 👍

willkg commented:
@thor You rock! I appreciate you fixing up `tox.ini` and `.travis.yml`.

This looks good to me!
